cyences_app_for_splunk/default/data/ui/views/cs_o365_reports.xml

```diff
@@ -989,4 +991,29 @@
       </table>
     </panel>
   </row>
-</form>
+  <row>
+    <panel>
+      <title>OneDrive or SharePoint File Sharing with External User</title>
+      <table>
+        <search>
+          <query>`cs_o365` sourcetype="o365:management:activity" Workload IN (OneDrive, SharePoint) Operation IN ("SharingSet", "AccessRequestApproved", "SharingRestored", "AnonymousLinkCreated", "SharingInvitationCreated", "SharingInvitationUpdated", "AddedToSecureLink", "SharingPolicyChanged") NOT TargetUserOrGroupType IN ("Member", "SharePointGroup") 
+| stats values(Operation) AS Operations, values(EventData) AS EventData, values(ItemType) as ItemType,values(EventSource) as EventSource latest(_time) as LastTime earliest(_time) as FirstTime by SourceFileName, UserId 
+| `cs_human_readable_time_format(FirstTime)`
```

Comment:
Add user privilege info by adding this macro: | `cs_user_privilege_mapping(User)`
and add is_privileged_user field to the table command.